src/main/scala/components/memory-async.scala

```diff
@@ -0,0 +1,378 @@
+// Asynchronous memory module
+
+package dinocpu
+
+import chisel3._
+import chisel3.util._
+
+import chisel3.experimental.ChiselEnum
+
+import chisel3.util.experimental.loadMemoryFromFile
+import firrtl.annotations.MemoryLoadFileType
+
+// A Bundle used for representing a memory access by instruction memory or data memory.
+class Request extends Bundle {  
```

Comment:
Can we parameterize the request with an access size? Maybe this is good for a follow-on commit.

Reply:
By access size, I assume you mean the width of the address, correct?

Reply:
No, I mean the size of the "block". For instance, right now we default to a 32-bit (4 byte) granularity of accessing memory. However, most caches have a block size of 64 bytes. 

At some point, I think we'll want to generalize this. However, it's not very high on my to do list right now :). We can just create an issue about it so we don't forget.